sui/typescript/dapp_kit_example/src/server/main.ts

```diff
@@ -42,6 +42,15 @@ const CREATE_WALLET_IF_NOT_FOUND = true;
 const WALLET_ONE_SUI_ADDRESS = await signer.getAddress(CREATE_WALLET_IF_NOT_FOUND);
 console.log("Invisible wallet address:", WALLET_ONE_SUI_ADDRESS.toString());
 
+// To use in case we don't want to send the sender signature to the FE.
+// If you want to get a signature on the FE but not let the user submit the 
+//  transaction until after you perform additional checks and then submit the
+//  transaction on the BE, you can do this by not sending the sponsor signature
+//  to the FE. Of course, in production you'd likely use a db where you store 
+//   other info alongside each signature, like the tx digest and tx bytes.
+const KEEP_SENDER_SIGNATURE_ON_BE = false; // must be set to false if the FE is submitting the tx
+let backendSponsorSignature: string | undefined = undefined;
```

Comment:
Don't share state between api calls through mutating global variables like this. It's immediately broken with more than one FE users. I couldn't really see a case where hiding the sponsor sig from FE is necessary. If additional checks are needed, why not do them BEFORE requesting sponsorship?

Reply:
Makes sense, and I updated the code to remove it. I agree that checking before makes the most sense. I just added a comment to say it's not technically needed to pass the signature the FE when the BE submits - anyone who comes along who believes they have a need to do a check after sender sig or wants to control exactly when a tx executes on the BE should be able to figure out how based on that.